### ISSUE TITLE
Need connection digagnostics

### ISSUE THREAD

**Issue by dhmspector** [SquaredTiki/Dringend-Issues#55]
Right now there's no way to debug a port-forwarding issue.  

I have 25 yrs experience designing & working with firewalls, my firewall is configured to allow the traffic on the default port, yet I am unable to get Drigend to connect to my build server.  (I know my settings are correct, my firewall allows other kinds of services thru.. but without any more diags from your app I have no way of knowing what's up).

**Comment by SquaredTiki** [comment]
Thanks for the suggestion. As for your particular issue: is port forwarding in The Constructor successful? Also are you able to successfully build using the private/local IP of your build server rather than the public IP?

**Comment by dhmspector** [comment]
The Constructor thinks the port forwarding works -- about to try locally to see if that works.  Works fine using a bonjour address...

**Comment by SquaredTiki** [comment]
Have you had success in getting this working?

**Comment by dhmspector** [comment]
Nope, not yet. Will get out the network sniffer next chance I get

## David

David HM Spector, Founder/Developer          Zeitgeist Info Systems          email: spector@zeitgeist.com  
m:415-480-0313     www.zeitgeist.com     twltter:@dhmspector      http://linkedin.com/in/davidhmspector

**Comment by scottsquires** [comment]
Same issue here. Getting a network time-out. The Constructor thinks the port forwarding is working fine. Compiling take a long time and then finally states host not responding without any diagnosis messages. Some type of progress monitor and some guidance would be great.
The port forwarding wasn't working earlier so I followed the faq but all it does it point to a 3rd party web site that lists hundreds of game and other software packages. More specifics about what specifically the app requires would be nice.

**Comment by SquaredTiki** [comment]
@dhmspector Ok, thank you.

@scottsquires So you had to manually set up port forwarding? The 3rd party website in the FAQ, portforward.com, has a number of guides for setting up port forwarding on a variety of routers. I'm not sure where you see lists of hundreds of games and software packages as when I follow the [link in the FAQ](http://portforward.com/english/routers/port_forwarding/routerindex.htm) there's a list of router brands (ignoring the adverts).

When manually port forwarding you'll need to know that The Constructor uses TCP and the port you'll want to forward is the one you are using in The Constructor, by default 5461.

Also, if you could both check that the port  you are trying to forward is open using [this website](http://www.yougetsignal.com/tools/open-ports/), that'd be great.

**Comment by dhmspector** [comment]
Solved my issue - it was an ISP routing issue (they were port blocking).

**Comment by SquaredTiki** [comment]
@dhmspector Glad to hear you solved it. If you don't mind sharing, how did you figure out in the end that it was an ISP issue? It would be useful for reference purposes so that others who see this can check too. Thanks.

**Comment by scottsquires** [comment]
> I'm not sure where you see lists of hundreds of games and software packages as when I follow the link in the FAQ there's a list of router brands (ignoring the adverts).

Follow that which then las a list of all the different routers from that particular company. Then follow that to a specific router and you'll see an entire long page of every game/app under the sun that you click on to configure for that specific one.

I have used their guide (guessing at which app best matches) and the guide for the router with still no success. Linksys router and Time Warner ISP

**Comment by SquaredTiki** [comment]
@scottsquires Ah I see what you mean now, where it is showing you the different services to configure for. Thanks for explaining.

What were the settings you used for the port forwarding?

Also from what @dhmspector said it might be worth checking if your ISP is port blocking. Have you checked using [this](http://www.yougetsignal.com/tools/open-ports/) that the port is open?

**Comment by scottsquires** [comment]
It confirmed that it's not actually forwarding despite the Linkysys settings. Since I have 2 routers involved and contacting TWC is a wasted day I'll have to back burner this for the moment.

**Comment by SquaredTiki** [comment]
I see, if you need any more assistance do just comment again and I'll be happy to help.
